docker/docker.go

```diff
@@ -101,6 +101,35 @@ func DoRunContainer(ctx context.Context, dClient *client.Client, imageCfg *conta
 	return nil
 }
 
+func DoRunOnetimeContainer(ctx context.Context, dClient *client.Client, imageCfg *container.Config, hostCfg *container.HostConfig, containerName string, hostname string, plane string, prsMap map[string]v3.PrivateRegistry) error {
+	if dClient == nil {
+		return fmt.Errorf("[%s] Failed to run container: docker client is nil for container [%s] on host [%s]", plane, containerName, hostname)
+	}
+	_, err := dClient.ContainerInspect(ctx, containerName)
+	if err != nil {
+		if !client.IsErrNotFound(err) {
+			return err
+		}
+		if err := UseLocalOrPull(ctx, dClient, hostname, imageCfg.Image, plane, prsMap); err != nil {
+			return err
+		}
+		resp, err := dClient.ContainerCreate(ctx, imageCfg, hostCfg, nil, containerName)
+		if err != nil {
+			return fmt.Errorf("Failed to create [%s] container on host [%s]: %v", containerName, hostname, err)
+		}
+		if err := dClient.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
+			return fmt.Errorf("Failed to start [%s] container on host [%s]: %v", containerName, hostname, err)
+		}
+		log.Infof(ctx, "Successfully started [%s] container on host [%s]", containerName, hostname)
+		log.Infof(ctx, "Waiting for [%s] container to exit on host [%s]", containerName, hostname)
+		if _, err := WaitForContainer(ctx, dClient, hostname, containerName); err != nil {
+			return fmt.Errorf("Container [%s] did not complete in time on host [%s]", containerName, hostname)
+		}
+		return nil
```

Comment:
If we are going to differentiate this, we should probably remove the container here as well if it served its purpose